src/components/BottomModal/BottomModal.js

```diff
@@ -0,0 +1,190 @@
+import React from 'react';
+import './BottomModal.scss';
+
+class BottomModal extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      isRequired: false,
+      isSelected: false,
+      selectedItems: [],
+    };
+  }
+
+  handleToggleClick = e => {
+    const {
+      target: { name },
+    } = e;
+    this.setState({
+      [name]: !this.state[name],
+    });
+  };
+
+  clickItems = e => {
+    const { isSelected, isRequired, selectedItems } = this.state;
+    const {
+      target: {
+        dataset: { name, price },
+      },
+    } = e;
+
+    if (isRequired || isSelected) {
+      this.setState({ isRequired: false, isSelected: false });
+    }
+
+    const obj = {
+      id: selectedItems.length,
+      name,
+      price,
+      quantity: 1,
+    };
+    const addItemToSelectedItems = selectedItems.concat(obj);
+    this.setState({ selectedItems: addItemToSelectedItems });
+  };
+
+  countQuantity = idx => {
+    const { selectedItems } = this.state;
+    console.log(selectedItems[idx]);
+  };
+
+  render() {
+    const { isRequired, isSelected, selectedItems } = this.state;
+    const { requireOption, selectOption, toggleModal } = this.props;
+
+    return (
+      <div className="outer">
+        <div className="overlay" onClick={toggleModal}></div>
+        <section className="BottomModal">
+          <section className="modal-header">
+            <p></p>
+            <p className="header-title">옵션선택</p>
+            <button className="close-button">
+              <img alt="close" src="/icon/Close.svg" onClick={toggleModal} />
+            </button>
+          </section>
+          <section className="modal-option">
+            <section className="required-option">
+              <p className="required-title">필수옵션</p>
+              <button
+                className="required-trigger"
+                name="isRequired"
+                onClick={e => this.handleToggleClick(e)}
```

Comment:
- event 는 기본적으로 인자로 넘겨지기 때문에 따로 넘겨주실 필요가 없습니다!
- 그리고 이렇게 되면, 렌더 메서드가 호출될 때 마다, `e => this.handleToggleClick(e)` 라는 함수가 계속해서 불필요하게 생성됩니다.
```suggestion
                onClick={this.handleToggleClick}
```

Reply:
수정 완료입니다🤗